service/src/main/java/org/mskcc/smile/service/impl/TempoServiceImpl.java

```diff
@@ -69,6 +72,7 @@ public Tempo getTempoDataBySamplePrimaryId(String primaryId) throws Exception {
     }
 
     @Override
+    @Transactional(rollbackFor = {Exception.class})
```

Comment:
My understanding of this annotation is that if some Exception occurs from this function (`mergeBamCompleteEventBySamplePrimaryId`), then any corresponding data changes will be rolled back as if this function wasn't called. That makes sense as that would keep the transactions atomic.

So I understand -- if we don't include this annotation, would whatever partial data changes that happen before the Exception occurs be kept?

Reply:
Yep, that's exactly it

Reply:
Thanks!